### PR TITLE
[INTEL MKL] Removed unnecessary omp_set_num() calls

### PR DESCRIPTION
**Pull request by vishakha-nervana**
Removed extra omp_set_num() calls and added call_once to ensure one time setting for OMP_NUM_THREADS.

### CODE REVIEW

File: tensorflow/core/common_runtime/threadpool_device.cc [hunk 1]
```diff
@@ -55,18 +54,14 @@ ThreadPoolDevice::ThreadPoolDevice(const SessionOptions& options,
...
     // OMP_NUM_THREADS controls MKL's intra-op parallelization
     // Default to available physical cores
     const int mkl_intra_op = port::NumSchedulableCPUs();
     const int ht = port::NumHyperthreadsPerCore();
-    omp_set_num_threads((mkl_intra_op + ht - 1) / ht);
-  } else {
-    uint64 user_val = 0;
-    if (strings::safe_strtou64(user_omp_threads, &user_val)) {
-      // Superflous but triggers OpenMP loading
-      omp_set_num_threads(user_val);
```
Comment (penpornk) [inline tensorflow/core/common_runtime/threadpool_device.cc:68]: If we remove `else`, we will lose OpenMP-pre-loading (according to the comment) in the case where users set the environment variable. Is this the desired behavior?

cc: @agramesh1

**Review by penpornk — COMMENTED** [review]
Thank you for the PR! I have a quick question.

File: tensorflow/core/common_runtime/threadpool_device.cc (hunk 1 above)
Comment (agramesh1) [reply tensorflow/core/common_runtime/threadpool_device.cc:68]: Hi @penpornk yes that is fine. We don't see a need to preload the library.

**Review by penpornk — APPROVED** [review]
Thank you for the quick confirmation!

**Review by gbaned — APPROVED** [review]

**Pull request merged**